pkg/container-utils/runtime-client/runtimeclient_test.go

```diff
@@ -0,0 +1,165 @@
+		t.Run(runtime.String(), func(t *testing.T) {
+			runtime := runtime
+			t.Parallel()
+
+			// Create test containers and their expected data
+			var expectedData []*runtimeclient.ContainerDetailsData
+			for i := 0; i < numContainers; i++ {
+				cn := fmt.Sprintf("%s-%s-%d", containerNamePrefix, runtime, i)
+				c, err := testutils.NewContainer(runtime, cn, "sleep inf")
+				require.Nil(t, err)
+				require.NotNil(t, c)
+				c.Start(t)
+				t.Cleanup(func() {
+					c.Stop(t)
+				})
+
+				expectedData = append(expectedData,
+					&runtimeclient.ContainerDetailsData{
+						ContainerData: runtimeclient.ContainerData{
+							Runtime: runtimeclient.RuntimeContainerData{
+								BasicRuntimeMetadata: types.BasicRuntimeMetadata{
+									RuntimeName:   runtime,
+									ContainerName: cn,
+									ContainerID:   c.ID(),
+								},
+								State: runtimeclient.StateRunning,
+							},
+							K8s: runtimeclient.K8sContainerData{},
+						},
+						Pid: c.Pid(),
+						// TODO: Is it worth to compare the cgroups path and mounts?
+					},
+				)
+			}
+
+			// Initialize runtime client
+			config := &containerutils.RuntimeConfig{
+				Name: runtime,
+			}
+			rc, err := containerutils.NewContainerRuntimeClient(config)
+			t.Cleanup(func() {
+				if rc != nil {
+					rc.Close()
+				}
+			})
+			require.Nil(t, err)
+			require.NotNil(t, rc)
+
+			// Test runtime client methods
+			t.Run("GetContainers", func(t *testing.T) {
+				t.Parallel()
+
+				containers, err := rc.GetContainers()
+				require.Nil(t, err)
+				require.NotNil(t, containers)
+
+				for _, eData := range expectedData {
+					found := false
+					for _, cData := range containers {
+						if cmp.Equal(*cData, eData.ContainerData) {
+							found = true
+							break
+						}
+					}
+					require.True(t, found, "couldn't find container:\n%s\nin:\n%s",
+						spew.Sdump(eData.ContainerData), spew.Sdump(containers))
+				}
+			})
+
+			t.Run("GetContainer", func(t *testing.T) {
+				t.Parallel()
+
+				for _, eData := range expectedData {
+					cData, err := rc.GetContainer(eData.Runtime.ContainerID)
+					require.Nil(t, err)
+					require.NotNil(t, cData)
+					require.True(t, cmp.Equal(*cData, eData.ContainerData),
+						"unexpected container data:\n%s", cmp.Diff(*cData, eData.ContainerData))
+				}
+			})
+
+			t.Run("GetContainerDetails", func(t *testing.T) {
+				t.Parallel()
+
+				for _, eData := range expectedData {
+					cData, err := rc.GetContainerDetails(eData.Runtime.ContainerID)
+					require.Nil(t, err)
+					require.NotNil(t, cData)
+
+					// TODO: Is it worth to compare the cgroups path and mounts?
+					require.NotEmpty(t, cData.CgroupsPath)
+					cData.CgroupsPath = eData.CgroupsPath
+					cData.Mounts = eData.Mounts
+
+					require.True(t, cmp.Equal(cData, eData),
+						"unexpected container data:\n%s", cmp.Diff(cData, eData))
+				}
+			})
+		})
+	}
+}
+
+// TestDeleteContainers is useful to delete containers from
```

Comment:
Is this a TODO or won't test

Reply:
I added it because during the development of the previous test, depending on where it failed, sometimes the environment remained dirty, so I used this function to enforce a cleanup. However, it is not an actual test. Maybe I should remove it?

Reply:
Ah, I see. Maybe this could then be even part of a cleanup function if a container is not deleted correctly (and then the test fails)

Reply:
It is already part of the cleanup function of the main test: `TestRuntimeClientInterface`, but during development, sometimes the test crashes and such function wasn't called. At that point, I executed this code ("test") to clean up all the remaining containers.